src/Platform.Handlers/src/Xamarin.Platform.Handlers/PropertyMapper.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xamarin.Platform
+{
+	public class PropertyMapper {
+		internal Dictionary<string, (Action<IViewRenderer, IFrameworkElement> Action, bool RunOnUpdateAll)> genericMap = new Dictionary<string, (Action<IViewRenderer, IFrameworkElement> action, bool runOnUpdateAll)> ();
+		protected virtual void UpdateProperty (string key, IViewRenderer viewRenderer, IFrameworkElement virtualView)
+		{
+			if (genericMap.TryGetValue(key, out var action))
+			{
+				action.Action?.Invoke(viewRenderer, virtualView);
+			}
+		}
+		public void UpdateProperty (IViewRenderer viewRenderer, IFrameworkElement virtualView, string property)
+		{
+			if (virtualView == null)
+				return;
+			UpdateProperty (property, viewRenderer, virtualView);
+		}
+		public void UpdateProperties (IViewRenderer viewRenderer, IFrameworkElement virtualView)
+		{
+			if (virtualView == null)
+				return;
+			foreach (var key in Keys) {
+				UpdateProperty (key, viewRenderer, virtualView);
+			}
+		}
+		public virtual ICollection<string> Keys => genericMap.Keys;
+	}
+
+	public class PropertyMapper<TVirtualView> : PropertyMapper, IEnumerable
```

Comment:
is there a use case for this to be subclassed (with part of the API being internal) ?